ndms2_client/client.py

```diff
@@ -194,6 +201,13 @@ def get_associated_devices(self):
 
         return devices
 
+    def save_configuration(self):
+        self._connection.run_command(_SAVE_CONFIGURATION)
```

Comment:
Note that when "Fail-safe configuration mode" is enabled on the router `system configuration save` is not enough.

Without repeated `system configuration fail-safe keep-alive` the router will reboot. To finalize the changes one needs to call `system configuration fail-safe commit`

Reply:
In manual I found no way to determine if fail-safe mode is active or not.
We have to issue both commands or instead introduce a separate command to commit changes and allow user to decide.

Reply:
Or just leave it as is for now. I have no idea how popular this mode is.

Reply:
I think we can issue both commands, but it would be great to execute the commit like 10-30 seconds later, so that if the change in interfaces would lead to HA loose connection to the router it would have time for this to actually happen. Then the reboot will revert the changes.

Reply:
I've added a separate method `commit_failsafe_configuration()`.  This will allow to schedule it in hass 30 seconds later.
I don't know another way to achieve '30 seconds delay' without long-lasting async operation, which I would like to avoid.

Reply:
Great. I will try testing the changes this weekend using your HA draft. Then if everything goes well, I'll merge this and push the tag

Reply:
BTW, hass integration doesn't save changes right now. I added `save_configuration()` for future use.